### PR TITLE
Allow inline configure of options in UseDeveloperExceptionPage

### LINKED ISSUE #219: Allow inline configure of options in UseDeveloperExceptionPage
In most components you can use an action the set the options.
```       
             app.UseDeveloperExceptionPage(options=>options
                    {
                        SourceCodeLineCount=10;
                    });
```
instead of
```
app.UseDeveloperExceptionPage(new Microsoft.AspNet.Diagnostics.ErrorPageOptions() 
           {
             SourceCodeLineCount=10
          });
```

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #219

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @Tratcher \ @Eilon

File: src/Microsoft.AspNet.Diagnostics/DeveloperExceptionPage/DeveloperExceptionPageExtensions.cs [hunk 1]
```diff
@@ -24,23 +23,31 @@ public static IApplicationBuilder UseDeveloperExceptionPage(this IApplicationBui
...
             {
                 throw new ArgumentNullException(nameof(builder));
             }
 
+            if (configureOptions == null)
+            {
+                throw new ArgumentNullException(nameof(configureOptions));
+            }
+
+            var options = new ErrorPageOptions();
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Diagnostics/DeveloperExceptionPage/DeveloperExceptionPageExtensions.cs]: Did options class never renamed to the middleware?

**Comment by Tratcher** [comment]
:shipit:

**Comment by Eilon** [comment]
:shipit:

File: src/Microsoft.AspNet.Diagnostics/DeveloperExceptionPage/DeveloperExceptionPageExtensions.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Diagnostics/DeveloperExceptionPage/DeveloperExceptionPageExtensions.cs]: Should we rename these? Using this overload is already broken for end users, so we might make the most of it and change the type name to match the middleware name. Thoughts?

Comment (Tratcher) [inline src/Microsoft.AspNet.Diagnostics/DeveloperExceptionPage/DeveloperExceptionPageExtensions.cs]: yes

**Pull request closed without merging**